docs/pages/guides/config-plugins.md

```diff
@@ -0,0 +1,614 @@
+export default createRunOncePlugin(
+  // The plugin to guard.
+  withMyCoolPlugin,
+  // An identifier used to track if the plugin has already been run.
+  pkg.name,
+  // Optional version property, if omitted, defaults to UNVERSIONED.
+  pkg.version
+);
+```
+
+## Debugging
+
+You can debug config plugins by running `EXPO_DEBUG=1 expo prebuild`. If `EXPO_DEBUG` is enabled, the plugin stack logs will be printed, these are useful for viewing which mods ran, and in what order they ran in. To view all static plugin resolution errors, enable `EXPO_CONFIG_PLUGIN_VERBOSE_ERRORS`, this should only be needed for plugin authors.
+By default some automatic plugin errors are hidden because they're usually related to versioning issues and aren't very helpful (i.e. legacy package doesn't have a config plugin yet).
+
+Running `expo prebuild --clean` with remove the generated native folders before compiling.
+
+You can also run `expo config --type prebuild` to print the results of the plugins with the mods unevaluated (no code is generated).
+
+Expo CLI commands can be profiled using `EXPO_PROFILE=1`.
+
+## Legacy plugins
+
+In order to make `eas build` work the same as the legacy `expo build` we added support for "legacy plugins" which are applied automatically to a project when they're installed in the project.
+
+For instance, say a project has `expo-camera` installed but doesn't have `plugins: ['expo-camera']` in their app.json. Expo CLI would automatically add `expo-camera` to the plugins so the camera and microphone permissions were always added to the project. The user can still customize the `expo-camera` plugin by adding it to the `plugins` array manually, the manually defined plugins will take precedence over the automatic plugins.
+
+You can debug which plugins were added by running `expo config --type prebuild` and seeing the `_internal.pluginHistory` property.
+
+This will show an object with all plugins that were added using `withRunOnce` plugin from `@expo/config-plugins`.
+
+Notice that `expo-location` uses `version: '11.0.0'`, and `react-native-maps` uses `version: 'UNVERSIONED'`. This means the following:
+
+- `expo-location` and `react-native-maps` are both installed in the project.
+- `expo-location` is using the plugin from the project's `node_modules/expo-location/app.plugin.js`
+- The version of `react-native-maps` installed in the project doesn't have a plugin, so it's falling back on the unversioned plugin that is shipped with `expo-cli` for legacy support.
+
+```js
+{
+  _internal: {
+    pluginHistory: {
+      'expo-location': {
+        name: 'expo-location',
+        version: '11.0.0',
+      },
+      'react-native-maps': {
+        name: 'react-native-maps',
+        version: 'UNVERSIONED',
+      },
+    },
+  },
+};
+```
+
+For the most _stable_ experience, you should try to have no `UNVERSIONED` plugins in your project. This is because the UNVERSIONED plugin may not support the native code in your project.
```

Comment:
```suggestion
For the most _stable_ experience, you should try to have no `UNVERSIONED` plugins in your project. This is because the `UNVERSIONED` plugin may not support the native code in your project.
```